src/app/multi-editor/multi-editor.component.ts

```diff
@@ -1,36 +1,117 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input, Output, OnInit, EventEmitter, ElementRef } from '@angular/core';
 import { ApiService } from '../shared/services/api.service';
 import { SchemaKeysStoreService } from '../shared/services/schema-keys-store.service';
 import { Observable } from 'rxjs';
+import { Headers, Http } from '@angular/http';
 
 @Component({
   selector: 'multi-editor',
   templateUrl: 'multi-editor.component.html',
-  styleUrls: ['multi-editor.component.scss']
+  styleUrls: ['multi-editor.component.scss'],
 })
-export class MultiEditorComponent implements OnInit {
 
+export class MultiEditorComponent implements OnInit {
+  currentPage = 1;
   records: Array<{}>;
+  totalRecords = 0;
   schema: {};
-
+  query: string;
+  queryUsed: string;
+  myRecord = {}
+  checkedRecords = [] //records that are different from the general selection rule
```

Comment:
make it map where key is `control_number ` and value is `boolean` for selected or not and when all selected/deselected just make everything true/false. It's more clear implementation on template and here.